tests/__tests__/html-transform.spec.ts

```diff
@@ -1,19 +1,32 @@
-import { process } from '../../src/preprocessor';
+import * as tsJest from '../../dist';
+import jestConfig from '../__helpers__/jest-config';
+
+const config = jestConfig.simple({});
+const filePath = `${config.rootDir}/some-file.html`;
+
+// wrap a transformed source so that we can fake a `require()` on it by calling the returned wrapper
+const wrap = (src: string) =>
+  new Function(`var module={}; ${src} return module.exports;`) as any;
 
 const source = `<div class="html-test">
   <span class="html-test__element">This is element</span>
+  <code>This is a backtilt \`</code>
 </div>`;
-const path = '/path/to/file.html';
-const config = {
-  globals: {
-    __TRANSFORM_HTML__: true,
-  },
-};
 
 describe('Html transforms', () => {
   it('transforms html if config.globals.__TRANSFORM_HTML__ is set', () => {
-    expect(process(source, path, config)).toMatchSnapshot();
-    delete config.globals.__TRANSFORM_HTML__;
-    expect(process(source, path, config)).toMatchSnapshot();
+    // get the untransformed version
+    const untransformed = tsJest.process(source, filePath, { ...config });
+    // ... then the one which should be transformed
+    (config.globals as any).__TRANSFORM_HTML__ = true;
+    const transformed = tsJest.process(source, filePath, config) as string;
+    // ... finally the result of a `require('module-with-transformed-version')`
+    const exported = wrap(transformed)();
+
+    expect(exported).toMatchSnapshot('module');
+    expect(transformed).toMatchSnapshot('source');
+    expect(untransformed).toMatchSnapshot('untransformed');
+    // requiring the transformed version should return the same string as the untransformed version
+    expect(exported).toBe(untransformed);
```

Comment:
I'm not very confident in this area of the codebase, perhaps @kulshekhar could weigh in here. It seems good to me but I have no clue.

Reply:
The `__TRANSFORM_HTML__` set to `true` is to make an html file become a module returning the html code as a string. Before it wasn't using `JSON.stringify` to make the module source. I've just fixed that issue (actually I think the fix has been merged in another pr before). Here the test has been improved to ensure that the source of the module is correct, the result of `reuqire()` it is correct, and that it matches with the `readFile()` content.